examples/wakaama/lwm2mclient.c

```diff
@@ -0,0 +1,698 @@
+        switch (lwm2mH->state) {
+            case STATE_INITIAL:
+                fprintf(stdout, "STATE_INITIAL\r\n");
+                break;
+            case STATE_BOOTSTRAP_REQUIRED:
+                fprintf(stdout, "STATE_BOOTSTRAP_REQUIRED\r\n");
+                break;
+            case STATE_BOOTSTRAPPING:
+                fprintf(stdout, "STATE_BOOTSTRAPPING\r\n");
+                break;
+            case STATE_REGISTER_REQUIRED:
+                fprintf(stdout, "STATE_REGISTER_REQUIRED\r\n");
+                break;
+            case STATE_REGISTERING:
+                fprintf(stdout, "STATE_REGISTERING\r\n");
+                break;
+            case STATE_READY:
+                fprintf(stdout, "STATE_READY\ren");
+                break;
+            default:
+                fprintf(stdout, "Unknown...\r\n");
+                break;
+        }
+        if (result != 0) {
+            fprintf(stderr, "lwm2m_step() failed: 0x%X\r\n", result);
+#ifdef LWM2M_BOOTSTRAP
+            if (previousState == STATE_BOOTSTRAPPING) {
+#ifdef WITH_LOGS
+                fprintf(stdout,
+                        "[BOOTSTRAP] restore security and server objects\r\n");
+#endif          /* WITH_LOGS */
+                prv_restore_objects(lwm2mH);
+                lwm2mH->state = STATE_INITIAL;
+            }
+            else
+#endif      /* LWM2M_BOOTSTRAP */
+            {
+                g_quit = 1;
+                continue;
+            }
+        }
+#ifdef LWM2M_BOOTSTRAP
+        DEBUG("update_bootstrap_info()\r\n");
+        update_bootstrap_info(&previousState, lwm2mH);
+#endif
+
+#ifdef MODULE_TINYDTLS
+handshake:
+#endif  /* MODULE_TINYDTLS */
+        DEBUG("xtimer_msg_receive_timeout64()\r\n");
+        //if (xtimer_msg_receive_timeout64(&msg, tv.tv_sec * 1000000) >= 0) {
+        pckt_rcvd_size = sock_udp_recv(&data.sock, &packet_rcvd, RCVD_SIZE, tv * US_PER_SEC, &remote);
+        if (pckt_rcvd_size > 0) {
+            connP = connection_find(data.connList, &remote);
+            if (connP != NULL) {
+                DEBUG("connection_handle_packet(%d)\r\n", pckt_rcvd_size);
+                int result = connection_handle_packet(connP, packet_rcvd, pckt_rcvd_size);
+                if (0 != result) {
+                    printf("error handling message %d\n", result);
+                }
+
+                /*
+                 * We loop around here until the DTLS handshake is done,
+                 * since lwm2m_step() will try to send stuff before then and fail
+                 * without retrying.
+                 */
+#ifdef MODULE_TINYDTLS
+                if (!connP->noSec) {
+                    dtls_peer_t *peer =
+                        dtls_get_peer(connP->dtlsContext, connP->dtlsSession);
+                    if (peer->state != DTLS_STATE_CONNECTED) {
+                        goto handshake;
+                    }
+                }
+#endif          /* MODULE_TINYDTLS */
+            }
+        }
+#ifdef MODULE_TINYDTLS
```

Comment:
I would recommend to remove this ifdef. 

As far as I know, is not breaking anything, but if you put this because the risk of losing packets while the handshake is taking place, then it's better to play with `DTLS_DEFAULT_MAX_RETRANSMIT`. 

Also, there is not logic in reseting the DTLS channel and then check if lwm2mH is in `STATE_READY`